### PR TITLE
#1739 provide the traceparent header as MDC value in logs

### PR DESCRIPTION
**Pull request by thjaeckle**
* generify the fields to provide to the MDC in CommonMdcEntryKey enum
* rename "x-correlation-id" in logs to just "correlation-id"
* exchange some places where only correlationId was extracted from a map of headers with parsing all the headers for MDC worthy fields

Resolves: #1739

### CODE REVIEW

**Comment by thjaeckle** [comment]
@alstanchev this PR would also be cool to be included in 3.4.0
I mentioned it briefly in today's Ditto community meeting, it provides the `traceparent` of [W3C Trace Context](https://www.w3.org/TR/trace-context/#traceparent-header) as MDC value in a generic way - similar to providing the `correlationId`.

I could simplify the loggers also quite a bit.

I also fixed that the `traceparent` header coming via an HTTP request was not used/forwarded.

**Comment by alstanchev** [comment]
> #1739

I have mentioned the issue https://github.com/eclipse-ditto/ditto/issues/1739 or do you want to mention the PR specifically?

**Comment by thjaeckle** [comment]
> I have mentioned the #1739 issue or do you want the PR to be mentioned specifically?

When there is an issue and a PR I normally only mention and link the issue ;) 

I also pinged you about reviewing, but you're right .. release notes would also be good - I could also do that however..

**Comment by thjaeckle** [comment]
@alstanchev @kalinkostashki Could you please have a look? Still missing an approval here .. :)

**Comment by alstanchev** [comment]
i am on it

File: gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/RootRouteHeadersStepBuilder.java [hunk 1]
```diff
@@ -122,7 +124,16 @@ QueryParametersStep withRequestContext(final RequestContext requestContext) {
 
         private Map<String, String> getFilteredExternalHeaders(final HttpMessage httpRequest) {
             final Iterable<HttpHeader> headers = httpRequest.getHeaders();
+            final Set<String> headerNames = new HashSet<>();
             final Map<String, String> externalHeaders = StreamSupport.stream(headers.spliterator(), false)
+                    .map(httpHeader -> {
```
Comment (alstanchev) [inline gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/RootRouteHeadersStepBuilder.java:129]: Just out of curiocity, what is the reason behind this double duplicate headers check?

**Review by alstanchev — APPROVED** [review]
LGTM

**Pull request merged**